src/System.Private.CoreLib/shared/System/Text/EncodingInfo.cs

```diff
@@ -2,70 +2,35 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Text;
-
 namespace System.Text
 {
     public sealed class EncodingInfo
     {
-        private int iCodePage;          // Code Page #
-        private string strEncodingName;    // Short name (web name)
-        private string strDisplayName;     // Full localized name
-
         internal EncodingInfo(int codePage, string name, string displayName)
         {
-            iCodePage = codePage;
-            strEncodingName = name;
-            strDisplayName = displayName;
-        }
-
-
-        public int CodePage
-        {
-            get
-            {
-                return iCodePage;
-            }
-        }
-
-
-        public string Name
-        {
-            get
-            {
-                return strEncodingName;
-            }
-        }
-
-
-        public string DisplayName
-        {
-            get
-            {
-                return strDisplayName;
-            }
+            CodePage = codePage;
+            Name = name;
+            DisplayName = displayName;
         }
 
+        public int CodePage { get; }
+        public string Name { get; }
+        public string DisplayName { get; }
 
         public Encoding GetEncoding()
         {
-            return Encoding.GetEncoding(iCodePage);
+            return Encoding.GetEncoding(CodePage);
         }
 
         public override bool Equals(object value)
         {
             EncodingInfo that = value as EncodingInfo;
-            if (that != null)
-            {
-                return (this.CodePage == that.CodePage);
-            }
-            return (false);
+            return CodePage == that?.CodePage;
```

Comment:
This makes `CodePage == 0` to be Equal to null. It should not matter, but it is atypical pattern for `Equals(object)`. I would stick with the usual pattern like what was in the original.

Reply:
This does normal nullable equality check under the hood so 0 and null are not equal.

Reply:
Ah ok. Still, I think the usual pattern would be better (and likely a bit smaller code too).

Reply:
I'd prefer modern C# and the generated code is the same in this case. See https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgBgARYCMA3ALABQuBhAdACpwAeMZ5FWAzAQEx4DCeAN4U8oglywAWPAFkAFAEohIsQF8VojdQBseYAHt9AGzwNYAOTgB3OQFEAdgGN9AEwCW9gOYBJewDN9PDg3JDwPGDwANwgjAFc4BS1hcjFUggB2ILcAfho+VzgABQhPODwAXnKomPjWVPU2FLEiXQNjUzhYAHkjFzsnVw8ff0Dg0PDquISkrVSsTOC8AEIq+1ijEwAyTay8guLSiqroqbq1ClUgA===

Reply:
I prefer code that is easy to understand and see that it is correct. I think the new version is worse on this metric.

I do not really care whether it is using modern C# or not.

Reply:
ok, will undo the change